tests/integration/command_test.py

```diff
@@ -0,0 +1,241 @@
+"""Various tests for `candies.cli.command.Command`."""
+
+from typing import Any
+
+import pytest
+
+from candies.cli.args.arg import Arg
+from candies.cli.args.opt import Opt
+from candies.cli.args.flag import Flag
+from candies.cli.cli import cli, CLI
+
+
+def execute(with_: str) -> Any:
+    try:
+        return cli(with_.split())
+    # `SystemExit`, because `argparse` calls `exit` on error.
+    except SystemExit as e:
+        return e
+
+
+@pytest.fixture(autouse=True)
+def reset():
+    # Reset `cli` before each test
+    # as if we have just imported it.
+    globals()['cli'] = CLI()
+
+
+def test_valid_command_name():
+    # Arrange.
+    @cli
+    def main():
+        pass
+
+    @main.command
+    def one():
+        pass
+
+    @main.command
+    def two():
+        pass
+
+    @one.command
+    def a():
+        return True
+
+    @one.command
+    def b():
+        return False
+
+    @two.command
+    def x():
+        return True
+
+    @two.command
+    def y():
+        return False
+
+    # Act.
+    cx_a = execute(with_='one a')
```

Comment:
I feel like it's probably would be better to reduce number of commands and checks in this test. In an ideal case, _Assert_ and _Act_ sections should occupy only one line each.

Maybe it's possible to rewrite the execution to only `cx = execute(with_='one a')` to check whether a sub-sub-command is called correctly?

Reply:
Why?